### PR TITLE
Update to babel v6 which should close #16.

### LINKED ISSUE #16: Update to Babel v6
https://github.com/babel/babel/blob/master/CHANGELOG.md

### PR DESCRIPTION
**Pull request by realistschuckle**
Update to babel v6 which should close #16.
Change timeout on mocha tests.
Use a newer version of mocha.
Use the new Travis build infrastructure.

### CODE REVIEW

File: test.js [hunk 1]
```diff
@@ -19,7 +20,20 @@ describe('Plugin', function() {
...
+    plugin = new Plugin({ plugins: { babel: { presets: [] }}});
+    plugin.compile({data: content, path: 'file.js'}, function (error, result) {
+      assert(!error);
+      assert(result.data.indexOf(content) !== -1);
+      done();
+    });
+  });
+
   it('should compile and produce valid result', function(done) {
+    this.timeout(10000);
```
Comment (es128) [inline test.js]: missed this one

Comment (realistschuckle) [inline test.js]: Sorry for the PITA I'm causing. I don't mean to take up so much of your time...

Comment (es128) [inline test.js]: Don't be sorry, there's no problem. I'm grateful you took the time to update the package.

**Comment by realistschuckle** [comment]
@es128 thanks for the help. Appreciate the guidance and input.

**Pull request merged**